api/test/unit/utils/validate_images_test.py

```diff
@@ -22,48 +21,26 @@ def get_redis_connection(*args, **kwargs):
     fake_redis.client().close()
 
 
-@dataclass
-class GRequestsFixture:
-    requests: list[AsyncRequest]
-    response_factory: Callable[
-        [AsyncRequest], Response
-    ] = lambda x: GRequestsFixture._default_response_factory(x)
-
-    @staticmethod
-    def _default_response_factory(req: AsyncRequest) -> Response:
-        res = Response()
-        res.url = req.url
-        res.status_code = 200
-        return res
-
-
-@pytest.fixture(autouse=True)
-def grequests(monkeypatch) -> GRequestsFixture:
-    fixture = GRequestsFixture([])
-
-    def map_reqs(reqs, **kwargs):
-        nonlocal fixture
-        fixture.requests += list(reqs)
-        responses = [fixture.response_factory(r) for r in fixture.requests]
-        return responses
-
-    monkeypatch.setattr("grequests.map", map_reqs)
-
-    return fixture
-
-
-def test_sends_user_agent(grequests):
+@mock.patch.object(aiohttp, "ClientSession", wraps=aiohttp.ClientSession)
+@pook.on
+def test_sends_user_agent(wrapped_client_session: mock.AsyncMock):
     query_hash = "test_sends_user_agent"
     results = [object() for _ in range(40)]
-    image_urls = [f"http://example.org/{i}" for i in range(len(results))]
+    image_urls = [f"https://example.org/{i}" for i in range(len(results))]
     start_slice = 0
 
+    head_mock = (
+        pook.head(pook.regex(r"https://example.org/\d"))
+        .times(len(results))
+        .reply(200)
+        .mock
+    )
+
     validate_images(query_hash, start_slice, results, image_urls)
 
-    requested_urls = [r.url for r in grequests.requests]
+    assert head_mock.calls > 0
```

Comment:
Would we want this to be an exact check? Is there a reason why some requests might fail?
```suggestion
    assert head_mock.calls == len(result)
```

Reply:
Nope, you're right, it should check exactly the number of calls :+1: